Purchases/Purchasing/Purchases.swift

```diff
@@ -1634,13 +1638,60 @@ public extension Purchases {
                                              observerMode: Bool,
                                              userDefaults: UserDefaults?,
                                              useStoreKit2IfAvailable: Bool) -> Purchases {
+        configure(
+            withAPIKey: apiKey,
+            appUserID: appUserID,
+            observerMode: observerMode,
+            userDefaults: userDefaults,
+            useStoreKit2IfAvailable: useStoreKit2IfAvailable,
+            dangerousSettings: nil
+        )
+    }
+
+    /**
+     * Configures an instance of the Purchases SDK with a custom userDefaults.
+     *
+     * Use this constructor if you want to sync status across a shared container,
+     * such as between a host app and an extension.
+     * The instance of the `Purchases` SDK will be set as a singleton.
+     * You should access the singleton instance using ``Purchases/shared``
+     *
+     * - Parameter apiKey: The API Key generated for your app from https://app.revenuecat.com/
+     *
+     * - Parameter appUserID: The unique app user id for this user. This user id will allow users to share their
+     * purchases and subscriptions across devices. Pass `nil` or an empty string if you want ``Purchases``
+     * to generate this for you.
+     *
+     * - Parameter observerMode: Set this to `true` if you have your own IAP implementation and want to use only
+     * RevenueCat's backend. Default is `false`.
+     *
+     * - Parameter userDefaults: Custom `UserDefaults` to use
+     *
+     * - Parameter dangerousSettings: Only use if suggested by RevenueCat support team.
+     *
+     * - Parameter useStoreKit2IfAvailable: EXPERIMENTAL. opt in to using StoreKit 2 on devices that support it.
+     * Purchases will be made using StoreKit 2 under the hood automatically.
+     * - Important: Support for purchases using StoreKit 2 is currently in an experimental phase.
+     * We recommend setting this value to `false` (default) for production apps.
+     *
+     * - Returns: An instantiated ``Purchases`` object that has been set as a singleton.
+     */
+    @objc(configureWithAPIKey:appUserID:observerMode:userDefaults:dangerousSettings:useStoreKit2IfAvailable:)
+    // swiftlint:disable:next function_parameter_count
+    @discardableResult static func configure(withAPIKey apiKey: String,
+                                             appUserID: String?,
+                                             observerMode: Bool,
+                                             userDefaults: UserDefaults?,
+                                             useStoreKit2IfAvailable: Bool,
+                                             dangerousSettings: DangerousSettings?) -> Purchases {
```

Comment:
Version 3 has another configure function with `configureWithAPIKey:appUserID:observerMode:userDefaults:dangerousSettings:`  https://github.com/RevenueCat/purchases-ios/pull/1219/files#diff-13e4912c09453a23a93ceace7903bcddb040d0db158cb39f0fffb1f1a3c6f1e4R180 . Do we need that configure version? I asume mostly no one is using this DangerousSettings so I think I prefer breaking v3 and give preference to `useStoreKit2IfAvailable` over `dangerousSettings`.

Reply:
Would adding a default value for the API `dangerousSettings: DangerousSettings? = nil` make it better? You could also create two APIs, one with `dangerousSettings` and one without with some combination of default values.

Reply:
Adding `dangerousSettings: DangerousSettings? = nil` doesn't fix it for objc. I thought about having two:

New function:
```
userDefaults: UserDefaults?,
useStoreKit2IfAvailable: Bool,
dangerousSettings: DangerousSettings?)
```

One matching v3:
```
userDefaults: UserDefaults?,
dangerousSettings: DangerousSettings?)
```

And the one we already have in v4
```
userDefaults: UserDefaults?,
useStoreKit2IfAvailable: Bool)
```

Is it worth it though?

Reply:
True, just because we can, should we? Probably not. The impact on our customers is probably extremely small.

Reply:
Yeah I would vote for that. The designated initializer should have all the parameters, and the convenience ones provide defaults. This is breaking that pattern, we don't wanna have multiple designated initializers. Thankfully Swift enforces that but not for `@objc` classes